### ISSUE TITLE
[portlayer] rm should do shortest prefix match for VM names

### ISSUE THREAD

**Issue by caglar10ur** [vmware/vic#1033]
**VIC Version:**

master

**Code**

```ts
[vagrant@devbox:/opt/go/src/github.com/vmware/vic(master)] $Z rm 14
Error response from daemon: No such container: 14
[vagrant@devbox:/opt/go/src/github.com/vmware/vic(master)] $Z rm 148e86e8327c9e4c3a21f98074a1ec53d2bc79547340877b8f349259aca1475b
148e86e8327c9e4c3a21f98074a1ec53d2bc79547340877b8f349259aca1475b
[vagrant@devbox:/opt/go/src/github.com/vmware/vic(master)]
```

**Expected behavior:**

First invocation should work and match

**Actual behavior:**

First invocation fails.

**Comment by jzt** [comment]
See:
https://github.com/docker/docker/blob/master/pkg/truncindex/truncindex.go
https://github.com/tchap/go-patricia

**Comment by hickeng** [comment]
also see https://github.com/vmware/vic/pull/1141#discussion_r67605934

**Comment by anchal-agrawal** [comment]
Can't reproduce this anymore, this was fixed in #1333.
